### PR TITLE
Add support of the CLUSTER REPLICAS command

### LINKED ISSUE #1942: Tracking Issue: Towards a more redis-compatible database
In order to be more compatible with redis, here are commands that exist in Redis (excluding Redis Stack) but are not supported in Kvrocks:

- [x] #1944
- [x] #1945
- [x] #1946
- [x] #1947
- [x] #1948
- [ ] #1949
- [x] #1950
- [x] #1951
- [x] #1952
- [x] #1954
- [x] #1955
- [x] #1958
- [x] #1959
- [x] #1983
- [x] #1984
- [x] #1985
- [x] #2166 
- [x] #2168
- [x] #2185
- [ ] ... (maybe not fully listed)

We very much welcome anyone to claim the tasks and start to contribute to Kvrocks. If you need any help during the contribution process, you can leave a message under the corresponding issue.

Also, the parts of existing commands that are incompatible with redis are worthy of attention and welcome any contribution.

- [x] #1212
- [x] #2006
- [x] #2014
- [ ] ... adding more

It succeed #595.

### PR DESCRIPTION
**Pull request by proost**
close: [1942](https://github.com/apache/kvrocks/issues/1942)

Add `CLUSTER REPLICAS` command

### CODE REVIEW

File: src/cluster/cluster.cc [hunk 1]
```diff
@@ -469,6 +469,52 @@ Status Cluster::GetClusterNodes(std::string *nodes_str) {
   return Status::OK();
 }
 
+Status Cluster::GetReplicas(const std::string &node_id, std::string *replicas_str) {
```
Comment (git-hulk) [inline src/cluster/cluster.cc]: Can use the StatusOr here:
```C++
StatusOr<std::string> Cluster::GetReplicas(const std::string &node_id) {

 ...
 return replicas_desc;
}
```

**Comment by git-hulk** [comment]
Generally LGTM, only one comment is inline.

**Review by caipengbo — DISMISSED** [review]
LGTM

File: src/cluster/cluster.cc [hunk 2]
```diff
@@ -469,6 +469,52 @@ Status Cluster::GetClusterNodes(std::string *nodes_str) {
...
+    if (n == nodes_.end()) {
+      continue;
+    }
+
+    auto replica = n->second;
+
+    std::string node_str;
+    // ID, host, port
+    node_str.append(replica_id + " ");
+    node_str.append(fmt::format("{}:{}@{} ", replica->host, replica->port, replica->port + kClusterPortIncr));
```
Comment (PragmaTwice) [inline src/cluster/cluster.cc]:
```suggestion
    node_str.append(fmt::format("{} {}:{}@{} ", replica_id, replica->host, replica->port, replica->port + kClusterPortIncr));
```

File: src/cluster/cluster.cc [hunk 3]
```diff
@@ -469,6 +469,52 @@ Status Cluster::GetClusterNodes(std::string *nodes_str) {
...
+
+    // Flags
+    node_str.append("slave " + replica->master_id + " ");
```
Comment (PragmaTwice) [inline src/cluster/cluster.cc]:
```suggestion
    node_str.append(fmt::format("slave {} ", replica->master_id));
```

File: src/cluster/cluster.cc (hunk 2 above)
Comment (proost) [reply src/cluster/cluster.cc]: [c9088c2](https://github.com/apache/kvrocks/pull/2244/commits/c9088c2dd2e7ea5ab206fbb2a707da4b2e18d167)

File: src/cluster/cluster.cc (hunk 1 above)
Comment (proost) [reply src/cluster/cluster.cc]: [c9088c2](https://github.com/apache/kvrocks/pull/2244/commits/c9088c2dd2e7ea5ab206fbb2a707da4b2e18d167)

File: src/cluster/cluster.cc (hunk 2 above)
Comment (proost) [reply src/cluster/cluster.cc]: [e9726fb](https://github.com/apache/kvrocks/pull/2244/commits/e9726fb4cee176feba13b633818f7a4d1652ddeb)

File: src/cluster/cluster.cc (hunk 3 above)
Comment (proost) [reply src/cluster/cluster.cc]: [e9726fb](https://github.com/apache/kvrocks/pull/2244/commits/e9726fb4cee176feba13b633818f7a4d1652ddeb)

**Review by PragmaTwice — DISMISSED** [review]

**Review by PragmaTwice — APPROVED** [review]

**Review by caipengbo — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]

**Comment by git-hulk** [comment]
@proost Thanks for your contribution.

**Pull request merged**